ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+#
+# Arguments :
+#        fh : a FileEditor object
+#        config_param: parameter to update
+#        line_fmt : line format (see LINE_FORMAT_xxx constants)
+sub removeConfigLine () {
+  my $function_name = "removeConfigLine";
+  my $self = shift;
+
+  my $fh = shift;
+  unless ( $fh ) {
+    $self->error("$function_name: 'fh' argument missing");
+    return 1;
+  }
+  my $config_param = shift;
+  unless ( $config_param ) {
+    $self->error("$function_name: 'config_param' argument missing");
+    return 1;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing");
+    return 1;
+  }
+
+  # Build a pattern to look for.
+  my $config_param_pattern = $self->buildLinePattern($config_param,$line_fmt);
+
+  $self->debug(1,"$function_name: commenting out lines matching pattern >>>".$config_param_pattern."<<<");
+  # All matching lines must be commented out, except if they are already commented out.
+  # The code used is a customized version of FileEditor::replace() that lacks backreferences.
+  my @lns;
+  my $line_count = 0;
+  seek($fh, 0, SEEK_SET);
+  while (my $l = <$fh>) {
+    if ($l =~ qr/^$config_param_pattern/ && $l !~ qr/^\s*#/) {
+        $self->debug(2,"$function_name: commenting out matching line >>>".$l."<<<");
+        $line_count++;
+        push (@lns, '#'.$l);
+    } else {
+        push (@lns, $l);
+    }
+  }
+  if ( $line_count == 0 ) {
+    $self->debug(1, "$function_name: No line found matching the pattern");
+  } else {
+    $self->debug(1, "$function_name: $line_count lines commented out");
+  }
+  $fh->set_contents (join("", @lns));
+ 
+}
+
+
+# This function do the actual update of a configuration line after doing the final
+# line formatting based on the line format.
+#
+# Arguments :
+#        fh : a FileEditor object
+#        config_param: parameter to update
+#        config_value : parameter value (can be empty)
+#        line_fmt : line format (see LINE_FORMAT_xxx constants)
+#        multiple : if true, multiple lines with the same keyword can exist (D: false)
+sub updateConfigLine () {
+  my $function_name = "updateConfigLine";
+  my $self = shift;
+
+  my $fh = shift;
+  unless ( $fh ) {
+    $self->error("$function_name: 'fh' argument missing");
+    return 1;
+  }
+  my $config_param = shift;
+  unless ( $config_param ) {
+    $self->error("$function_name: 'config_param' argument missing");
+    return 1;
+  }
+  my $config_value = shift;
+  unless ( defined($config_value) ) {
+    $self->error("$function_name: 'config_value' argument missing");
+    return 1;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing");
+    return 1;
+  }
+  my $multiple = shift;
+  unless ( defined($multiple) ) {
+    $multiple = 0;
+  }
+
+  my $newline;
+  my $config_param_pattern;
+  $newline = $self->formatConfigLine($config_param,$config_value,$line_fmt);
```

Comment:
`my $newline = ..` and remove `my $newline` above.
also maybe use `new_line` or just `line` (as a `newline` is something else)